src/librustc_mir/interpret/intrinsics.rs

```diff
@@ -239,7 +239,25 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             "transmute" => {
                 self.copy_op_transmute(args[0], dest)?;
             }
-
+            "simd_insert" => {
+                let mut vector = self.read_vector(args[0])?;
```

Comment:
instead of reading, modifying and writing a vector, you could iterate over the vector in one pass. If the index matches the current position, write the scalar, if the index does not, use `operand_field` to get the field and write it directly to `dest` (via `place_field`) (instead of actually reading the operand to an `ImmTy` that is just going to get written back out)

Reply:
@oli-obk so I've tried that using `self.copy_op` to perform the copies, but I got errors that using the values are now uninitialized. I changed that to the current code, but still got these errors :/

Reply:
Ok never mind, found the issue I think, wrong indexing..